h2o-core/src/main/java/water/rapids/ast/prims/advmath/AstCorrelation.java

```diff
@@ -59,7 +62,40 @@ public Val apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
         throw new IllegalArgumentException("unknown use mode: " + use);
     }
 
-    return fry.numRows() == 1 ? scalar(frx, fry, mode) : array(frx, fry, mode);
+    final Method method = getMethodFromUserInput(stk.track(asts[4].exec(env)).getStr());
+    switch (method) {
+      case Pearson:
+        return fry.numRows() == 1 ? scalar(frx, fry, mode) : array(frx, fry, mode);
+      case Spearman:
+        return spearman(frx, fry, mode);
+      default:
+        throw new IllegalStateException(String.format("Given method input'%s' is not supported. Available options are: %s",
+                method, Arrays.toString(Method.values())));
+    }
+  }
+
+  private static Method getMethodFromUserInput(final String methodUserInput) {
```

Comment:
Thank you for catching this. The whole method's logic is wrong. I wrote the method separately to match in a case-insensitive way, but never really "implemented" it. There are multiple problems with this method.

1. I ignored my own code in `EnumUtils`, specifically method `valueOfIgnoreCase`, which does what I need.
2. We really need to match in a case-insensitive way. In R, the `cor` function uses all lower-case, while in Python, there are different styles. One solution is to convert the string on the client (Py/R/Rapids written directly ?), but this is a simpler and more centralized solution. I don't see a reason to punish our users for case-related mistakes. No trimming, no guessing, just case-related mistakes. We do this on many places.



Using EnumUtils instead. Also, made `EnumUtills.valueOfIgnoreCase` return Optional, which required some minor "enchantments" in non-related code (three occurances basically).

Reply:
The method is now so small it could be removed and it's content used directly, IDK. Matter of code style I guess. Looks more obvious from the main method I guess.